packages/deployments/contracts/contracts/lib/Connext/ConnextLogic.sol

```diff
@@ -644,22 +645,27 @@ library ConnextLogic {
     // make sure number of routers is valid
     if (pathLength > _args.maxRoutersPerTransfer) revert ConnextLogic__execute_maxRoutersExceeded();
 
+    // get transfer id
+    bytes32 transferId = _getTransferId(_args);
+
+    // get the payload the router should have signed
+    bytes32 routerHash = keccak256(abi.encode(transferId, pathLength));
+
     // make sure routers are all approved if needed
-    // TODO: check the signature of each of the routers within this loop
     if (!_args.isRouterOwnershipRenounced) {
       for (uint256 i; i < pathLength; ) {
         if (!_approvedRouters[_args.executeArgs.routers[i]]) {
           revert ConnextLogic__execute_notSupportedRouter();
         }
+        if (_args.executeArgs.routers[i] != _recoverSignature(routerHash, _args.executeArgs.routerSignatures[i])) {
+          revert ConnextLogic__execute_invalidRouterSignature();
+        }
```

Comment:
Shouldn't we be checking for the signatures also if `_args.isRouterOwnershipRenounced` is `true`?